modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ipfinder/vm/TcpDiscoveryVmIpFinder.java

```diff
@@ -214,40 +214,43 @@ else if (ipStr.indexOf(':') >= 0)
         if (tokens.length == 2) {
             String addrStr = tokens[0];
             String portStr = tokens[1];
+            int port1, port2;
 
             if (portStr.contains("..")) {
-                try {
-                    int port1 = Integer.parseInt(portStr.substring(0, portStr.indexOf("..")));
-                    int port2 = Integer.parseInt(portStr.substring(portStr.indexOf("..") + 2, portStr.length()));
+                port1 = Integer.parseInt(portStr.substring(0, portStr.indexOf("..")));
+                port2 = Integer.parseInt(portStr.substring(portStr.indexOf("..") + 2, portStr.length()));
+            } else
+                port1 = port2 = Integer.parseInt(portStr);
 
-                    if (port2 < port1 || port1 == port2 || port1 <= 0 || port2 <= 0)
-                        throw new IgniteSpiException(errMsg);
+            if (port1 != port2 && port2 < port1 || port1 <= 0 || port2 <= 0)
```

Comment:
Can you put the first two into ()? if ((port1 != port2 && port2 < port1) || port1 <= 0 || port2 <= 0) for better readability

Reply:
Fixed.